ReSharper.FSharp/src/FSharp.Common/src/Checker/FcsProjectProvider.fs

```diff
@@ -315,15 +320,26 @@ type FcsProjectProvider(lifetime: Lifetime, solution: ISolution, changeManager:
         member this.GetFcsProject(psiModule) =
             getOrCreateFcsProject psiModule
 
+        member this.GetProjectOutputPaths(project) =
+            seq { for psiModule in projectsPsiModules.[project] do
+                  match outputPaths.TryGetValue(psiModule) with
+                  | null -> ()
+                  | path -> yield path }
+
+
 /// Invalidates psi caches when a non-F# project is built and FCS cached resolve results become stale
 [<SolutionComponent>]
 type OutputAssemblyChangeInvalidator(lifetime: Lifetime, outputAssemblies: OutputAssemblies, daemon: IDaemon,
-        psiFiles: IPsiFiles, fcsProjectProvider: IFcsProjectProvider, scheduler: ISolutionLoadTasksScheduler) =
+        psiFiles: IPsiFiles, fcsProjectProvider: IFcsProjectProvider, scheduler: ISolutionLoadTasksScheduler,
+        typeProvidersShim: IProxyExtensionTypingProvider) =
     do
         scheduler.EnqueueTask(SolutionLoadTask("FSharpProjectOptionsProvider", SolutionLoadTaskKinds.StartPsi, fun _ ->
             // todo: track file system changes instead? This currently may be triggered on a project model change too.
             outputAssemblies.ProjectOutputAssembliesChanged.Advise(lifetime, fun (project: IProject) ->
-                if not fcsProjectProvider.HasFcsProjects || project.IsFSharp then () else
+                if not fcsProjectProvider.HasFcsProjects ||
+                   project.IsFSharp &&
+                   fcsProjectProvider.GetProjectOutputPaths(project)
+                   |> typeProvidersShim.HasGenerativeTypeProviders |> not then () else
```

Comment:
Would it be hard to cache it? We don't want it to be repeatedly called on big solutions.

Reply:
implemented memoization of calculations